### ISSUE TITLE
[長期訓練] GPU 擴展訓練計劃 - 目標 50-100GB 數據 + 24/7 訓練

### ISSUE THREAD

**Issue by christophert0607-creator** [christophert0607-creator/kiro-quant-v3#25]
# 🚀 GPU 長期訓練擴展計劃

## 📊 目標
- 數據量: 50-100GB parquet
- 訓練時間: 24/7 運行
- GPU: RTX 3060 (12GB VRAM)

---

### 1.1 新增數據目錄
```bash
# 創建新目錄
mkdir -p v3_pipeline/data/storage/base_20y/
mkdir -p v3_pipeline/data/storage/multi_stock_2026/
```

### 1.2 添加更多股票
需要添加 50-100 隻股票：
- HK: 00700 (騰訊), 09988 (阿里), 02800 (盈富) 等
- US: 持續添加更多標的

### 1.3 使用 futu_api 拉取數據
```python
from futu_api import get_historical_kline

symbols = ["00700", "09988", "02800", ...] # 添加 50-100 隻
for sym in symbols:
    df = get_historical_kline(sym, start="2010-01-01", end="2026-03-12", interval="1D")
    df.to_parquet(f"v3_pipeline/data/storage/base_20y/{sym}.parquet")
```

### 1.4 添加特徵工程
使用現有模組添加 20+ 特徵：
- RSI
- MACD
- 資金流
- 布林帶
- 等...

---

### 2.1 Mixed Precision + 省 VRAM
```python
from torch.cuda.amp import autocast, GradScaler

scaler = GradScaler()
with autocast(dtype=torch.bfloat16):  # RTX 3060 穩定支援
    loss = model(...)
```

### 2.2 PyTorch 2.0+ 加速
```python
model = torch.compile(model)  # 加速 20-30%
```

### 2.3 Gradient Accumulation
```python
# 有效 batch size = 32 * 8 = 256
batch_size = 32
accum_steps = 8
```

### 2.4 TF32 加速
```python
torch.backends.cuda.matmul.allow_tf32 = True
torch.backends.cudnn.allow_tf32 = True
```

### 2.5 Checkpoint 超頻
- 每 500 step 保存一次
- 添加 resume 功能（斷電後可接續）

---

### 3.1 使用 tmux（最穩定）
```bash
# 創建 tmux session
tmux new -s quant_pretrain

# 啟動訓練
python v3_pipeline/models/trainer_v4_1.py --data-path base_20y --epochs 50 --batch 64

# 離開 tmux
Ctrl+B D

# 重新連接
tmux attach -t quant_pretrain
```

### 3.2 監控腳本
```bash
# GPU 監控
nvidia-smi -l 10

# 訓練監控
python training_check.py
```

### 3.3 Streamlit Dashboard
```bash
streamlit run dashboard.py
```
可實時查看 loss curve + metrics

---

## 📋 執行順序

1. **Day 1**: 創建數據目錄 + 添加 50 隻股票數據
2. **Day 2**: 配置 trainer_v4_1.py 優化參數
3. **Day 3**: 啟動 24/7 訓練
4. **持續**: 監控 + 調整

---

## ✅ 需要完成的事項

- [ ] 創建數據目錄結構
- [ ] 添加股票列表 (50-100隻)
- [ ] 實現數據自動獲取腳本
- [ ] 添加特徵工程
- [ ] 優化 trainer_v4_1.py 參數
- [ ] 添加 mixed precision 支持
- [ ] 添加 checkpoint resume 功能
- [ ] 設置 tmux 訓練環境
- [ ] 配置監控腳本

---

## 🎯 預期效果
- 更強的預測模型
- 覆蓋更多股票
- 24/7 穩定運行
- RTX 3060 充分利用

@codex 請幫忙實現這個計劃！
